core/src/main/java/io/questdb/griffin/engine/functions/catalogue/FlushQueryCacheFunctionFactory.java

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.catalogue;
+
+import io.questdb.MessageBus;
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BooleanFunction;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+
+public class FlushQueryCacheFunctionFactory implements FunctionFactory {
+
+    private static final Log LOG = LogFactory.getLog("flush-query-cache");
+
+    @Override
+    public String getSignature() {
+        return "flush_query_cache()";
+    }
+
+    @Override
+    public Function newInstance(int position,
+                                ObjList<Function> args,
+                                IntList argPositions,
+                                CairoConfiguration configuration,
+                                SqlExecutionContext sqlExecutionContext
+    ) {
+        return new FlushQueryCacheFunction(sqlExecutionContext.getMessageBus());
```

Comment:
considering we have single Bus it makes sense to have 1-2-1 relationship between factory and function. Function instance can be factory member variable. The only thing to look out for is to take care of multi-threaded instance  access

Reply:
It's not that it's a hot path code, so personally I'd better keep the code simpler by allocating function instances. Also, embedded API users are free to spin up multiple `CairoEngine` instances (thus, message bus instances too), so we can't assume that there is a 1-1 relationship between factory and message bus (hence, function). Considering this, I'd better keep the code correct rather than optimize it without a strong reason. WDYT?

Reply:
it is a good point regarding multiple engines, we're likely to make a move in this direction as well